pkg/cmd/server/api/types.go

```diff
@@ -146,9 +148,120 @@ type OAuthConfig struct {
 	// AssetPublicURL is used for building valid client redirect URLs for external access
 	AssetPublicURL string
 
-	// all the handlers here
+	//IdentityProviders is an ordered list of ways for a user to identify themselves
+	IdentityProviders []IdentityProvider
+
+	// GrantConfig describes how to handle grants
+	GrantConfig GrantConfig
+
+	// SessionAuthenticationConfig hold information about configuring sessions.
+	SessionAuthenticationConfig *SessionAuthenticationConfig
+
+	TokenConfig TokenConfig
+}
+
+type TokenConfig struct {
+	// Max age of authorize tokens
+	AuthorizeTokenMaxAgeSeconds int32
+	// Max age of access tokens
+	AccessTokenMaxAgeSeconds int32
+}
+
+type SessionAuthenticationConfig struct {
+	// SessionSecrets list the secret(s) to use to encrypt created sessions. Used by AuthRequestHandlerSession
+	SessionSecrets []string
+	// SessionMaxAgeSeconds specifies how long created sessions last. Used by AuthRequestHandlerSession
+	SessionMaxAgeSeconds int32
+	// SessionName is the cookie name used to store the session
+	SessionName string
+}
+
+type IdentityProviderUsage struct {
+	// ProviderScope is used to qualify the identities teturned by this provider
+	ProviderScope string
+
+	// UseAsChallenger indicates whether to issue challenges for this provider
+	UseAsChallenger bool
+	// UseAsLogin indicates whether to use this identity provider to back the login screen
+	UseAsLogin bool
+}
+
+type IdentityProvider struct {
+	// Usage contains metadata about how to use this provider
+	Usage IdentityProviderUsage
+
+	// Provider contains the information about how to set up a specific identity provider
+	Provider runtime.EmbeddedObject
+}
+
+type BasicAuthPasswordIdentityProvider struct {
+	api.TypeMeta
+
+	// RemoteConnectionInfo contains information about how to connect to the external basic auth server
+	RemoteConnectionInfo RemoteConnectionInfo
+}
+
+type AllowAllPasswordIdentityProvider struct {
+	api.TypeMeta
+}
+
+type DenyAllPasswordIdentityProvider struct {
+	api.TypeMeta
 }
 
+type HTPasswdPasswordIdentityProvider struct {
+	api.TypeMeta
+
+	// File is a reference to your htpasswd file
+	File string
+}
+
+type XRemoteUserIdentityProvider struct {
+	api.TypeMeta
+
+	// CAFile is a file with the trusted signer certs
+	CAFile string
```

Comment:
Marker for me to come back to this.  I realized last night that this is what was pooching my test.